src/Microsoft.TemplateEngine.Edge/Template/TemplateMatchInfoEqualityComparer.cs

```diff
@@ -3,6 +3,7 @@
 
 namespace Microsoft.TemplateEngine.Edge.Template
 {
+    [Obsolete("The class is deprecated.")]
```

Comment:
nit: I also noticed `The` vs. ` This` like @bozturkMSFT here: https://github.com/dotnet/templating/pull/3013#discussion_r610917928
So I looked at what people are doing in .NET source code, and looks like its mostly "this": https://source.dot.net/#System.Private.CoreLib/ObsoleteAttribute.cs,d386ddc6e8f1405e,references
I think its fine to leave it as is for this PR, but we probably want to use `this` going foward

Reply:
Yeah, I agree this message is bad, but here and in other examples with this message the class is removed and there is no replacement. Not sure what to suggest in obsolete message...